frontend/src/components/Events.svelte

```diff
@@ -0,0 +1,246 @@
+<script lang="ts">
+import Modal from "./Modal.svelte";
+import CalendarPlusIcon from "../assets/calendar-plus.svg";
+import AlarmIcon from "../assets/alarm.svg";
+import GeoIcon from "../assets/geo.svg";
+import InfoIcon from "../assets/info_icon.svg";
+import { Browser } from "@wailsio/runtime";
+
+let {
+  visible = $bindable(false),
+  eventsNow = $bindable(0),
+  eventsFuture = $bindable(0),
+} = $props();
+
+let events: {
+  name: string;
+  location: string;
+  time: string;
+  timeUntil: string;
+  remainingTimeInMs: number;
+  moreInfo: string;
+  logo: string;
+}[] = $state([]);
+
+function dateTimeCheck(today: Date, event: any) {
+  const names = event.summary;
+  const start = event.start.dateTime;
+  const newDate = new Date(start);
+
+  if (event.recurrence) {
+    try {
+      const recurrence = event.recurrence[0].split(";");
+      const recType = recurrence[0].split("=")[1];
+      const interval = recurrence[2].split("=")[1];
+      const endDateType = recurrence[2].split("=")[0];
+      const endDateRaw = recurrence[2].split("=")[1];
+
+      let endDate = new Date(newDate);
+      if (endDateType === "COUNT") {
+        if (recType === "WEEKLY") {
+          endDate.setDate(newDate.getDate() + 7 * interval * endDateRaw);
+        } else if (recType === "MONTHLY") {
+          endDate.setDate(newDate.getDate() + 4 * interval * endDateRaw);
+        }
+      } else {
+        endDate.setDate(endDateRaw);
+      }
+
+      if (recType === "WEEKLY") {
+        while (newDate <= endDate) {
+          if (newDate > today) {
+            break;
+          }
+
+          newDate.setDate(newDate.getDate() + 7);
+        }
+      } else if (recType === "MONTHLY") {
+        while (newDate <= endDate) {
+          if (newDate > today) {
+            break;
+          }
+
+          newDate.setDate(newDate.getDate() + 4);
+        }
+      }
+    } catch (e) {
+      console.error(`Error checking recurrence: ${e}`);
+    }
+  }
+
+  const remainingTimeInMs = newDate.getTime() - today.getTime();
+  return [names, newDate, remainingTimeInMs];
+}
+
+function formatFromNow(milliseconds: number) {
+  const minuteMillis = 1000 * 60;
+  const hourMillis = minuteMillis * 60;
+  const dayMillis = hourMillis * 24;
+
+  let format = "";
+
+  const days = Math.floor(milliseconds / dayMillis);
+  if (days > 0) {
+    format += days;
+    format += days > 1 ? " days " : " day ";
+  }
+
+  const hours = Math.floor((milliseconds % dayMillis) / hourMillis);
+  if (hours > 0) {
+    format += hours;
+    format += hours > 1 ? " hours " : " hour ";
+  }
+
+  const minutes = Math.floor((milliseconds % hourMillis) / minuteMillis);
+  if (minutes > 0) {
+    format += minutes;
+    format += minutes > 1 ? " minutes " : " minute ";
+  }
+
+  return format;
+}
+
+function fetchEvents() {
```

Comment:
this should probably be async, and the `.then`s should probably be await

Reply:
Good point.